File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/JavaScript/_014)_Second_Largest_Number.js
function getSecondLargest(arr) {
    let res = -1,
        largest = 0;
    for (let i = 1; i < arr.length; i++) {
        if (arr[i] > arr[largest]) {
            res = largest;
            largest = i;
        } else if (arr[i] != arr[largest]) {
            if (res == -1 || arr[i] > arr[res]) res = i;
        }
    }
    return res;
}
let arr = [10, 5, 18, 20];
let index = getSecondLargest(arr);
console.log(`Index = ${index}
Second Largest Number in [${arr}] is ${arr[index]}`);

File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/Dart/_014)_Second_Largest_Number.dart
int getSecondLargest(arr) {
  var res = -1, largest = 0;
  for (var i = 1; i < arr.length; i++) {
    if (arr[i] > arr[largest]) {
      res = largest;
      largest = i;
    } else if (arr[i] != arr[largest]) {
      if (res == -1 || arr[i] > arr[res]) res = i;
    }
  }
  return res;
}

void main() {
  var arr = [10, 100, 20, 70, 40, 50, 60, 30, 80, 90];
  var index = getSecondLargest(arr);
  print('Second largest element in ${arr} is ${arr[index]}.');
}
